RobotCrawler/src/test/java/parser/ebookscom/EBooksParserTest.java

```diff
@@ -0,0 +1,99 @@
+package parser.ebookscom;
+
+import model.BookData;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+import parser.BookDataGetter;
+import parser.DocumentBuilder;
+import parser.Parser;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * This parser is responsible to parse data prom the
+ */
+public class EBooksParserTest {
+
+    private Document theHtmlFile = null;
+    private Parser<Element> parser = new EBooksParser();
+    private BookDataGetter getter = new EBooksDataGetter();
+    @BeforeTest
+    protected void initTest(){
+        File file = new File(System.getProperty("user.dir") + "/src/test/resources/parser-ebook-test-1.html");
+        try {
+            theHtmlFile = DocumentBuilder.builder().file(file).build().buildFromFile();
+        } catch (IOException e) {
+            // TODO there must be the logger
+            throw new AssertionError();
+        }
+    }
+
+    @Test
+    public void checkIfParserFindsOneOffersFromFile(){
+
+        // given - theHtmlFile
+
+        // when
+        List<Element> offers = parser.parse(theHtmlFile);
+
+        // then
+        if(offers==null){
+            // this state is invalid and test cannot be passed in this situation
```

Comment:
Interesting. Is this guarding against an unlikely condition, or is this a condition that - if happened - makes test run useless?